### PR TITLE
feat(vault): encode to HCL the vault path rule and add some tests

### PR DESCRIPTION
**Pull request by ccojocar**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

feat(vault): encode to HCL the vault path rule and add some tests

### CODE REVIEW

File: pkg/vault/rule_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		},
+		"marshal nil policy with error": {
+			rule: nil,
+			err:  true,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			output, err := tc.rule.String()
```
Comment (agentgonzo) [inline pkg/vault/rule_test.go]: Don't compare strings for things like this. 
1) It causes a lot of unnecessary headaches (eg, if we change a library implementation at some point, or change whitespace formatting, the tests will fail). Also, testing like this broke on Windows where a newline is CR-LF but in unix it's just LF
2) Ultimately, we don't care about the `string` representation, we care about the `json`/`yaml` representation.

keep `tc.rule` as a struct (or a JSON/YAML object) and deserialize the the `want` to the same type, then do a comparison on the typed objects.

**Comment by ccojocar** [comment]
@agentgonzo I modified the test to use the HCL parser, and now it compares the structs instead of the strings. Please could you have a look? Thanks

**Pull request merged**